src/drafts/Tooltip/Tooltip.tsx

```diff
@@ -32,44 +20,31 @@ const animationStyles = `
 `
 
 const StyledTooltip = styled.div`
-  /* tooltip element should be rendered visually hidden when it is not opened.  */
-  width: 1px;
-  height: 1px;
-  padding: 0;
-  margin: -1px;
-  overflow: hidden;
-  clip: rect(0, 0, 0, 0);
-  white-space: nowrap;
-  position: fixed;
-  font: normal normal 11px/1.5 ${get('fonts.normal')};
-  -webkit-font-smoothing: subpixel-antialiased;
-  color: ${get('colors.fg.onEmphasis')};
-  text-align: center;
-
-  word-wrap: break-word;
-  background: ${get('colors.neutral.emphasisPlus')}; //bg--emphasis-color
-  border-radius: ${get('radii.2')};
-  border: 0;
-  opacity: 0;
-  max-width: 250px;
-  inset: auto;
+  /* Overriding the default popover styles */
+  &[popover] {
```

Comment:
This was needed for firefox. Otherwise the styles wasn't applied due to `[popover]` having more specificity than the styled component's class name.

Reply:
This trick caused a bit of a flashing though. 

https://github.com/primer/react/assets/1446503/cde9f8cf-6ded-45ce-b1a9-59dc8b77e57e
_Video description: There is a "Delete" button and the page is reloaded. The tooltip text without any styles appears right under the button very briefly like flashing then disappears_

And if I load the style component (the tooltip element itself) with `display: 'none'` and then add `display: 'block'` when it is open, the flash is gone ([commit](https://github.com/primer/react/pull/3910/commits/7552c3bc4bf80a92898ba1222f31336f7991f3e6) for changes) but I am not sure if this is a good way to solve this. Let me know if you have any a way of solving this already or any other thoughts!

Reply:
This looks reasonable for now. The polyfill injecting CSS means that we suffer from a FOUC, which is also true for dotcom today. I'm looking into ways to minimise or remove the FOUC but given it's only Firefox, and Firefox will soon be shipping native popovers, it's not a blocking concern.

Reply:
> And if I load the style component (the tooltip element itself) with display: 'none' and then add display: 'block' when it is open

I kinda like this as a temporary measure but also if the flash only happens in firefox, we can choose to ignore it. No strong feelings, up to you

Reply:
Just to clarify, the fix that needed for firefox was adding the `[popover]` attribute selector to the class styling but this addition caused flashing in all browsers. So I guess it is better to keep it until Firefox catches up.